### ISSUE TITLE
Test server issues legacy query when resolving activity at the same time but cannot handle legacy query response

### ISSUE THREAD

**Issue by cretz** [temporalio/sdk-java#1428]
## Expected Behavior

Test server should not issue legacy query or handle legacy query response

## Actual Behavior

This one is a bit hard to reproduce. You have to make an activity resolution happen at the same time as a query. Then the query comes over as a legacy query and core responds to the legacy query causing:

> Lang SDK sent us a malformed workflow completion for run (a067d736-1f05-4217-8c4c-4c4c5674c605): Workflow completion had a legacy query response along with other commands. This is not allowed and constitutes an error in the lang SDK

**Comment by cretz** [comment]
Hrmm, this may be a core issue. Investigating.

**Comment by Spikhalskiy** [comment]
Test server issues legacy queries and is able to process legacy query responses. This is a pretty heavily tested path in java-sdk.

**Comment by cretz** [comment]
Yeah, this is core related or an expectation that legacy queries and other events don't come over in the same task. Checking...

**Comment by cretz** [comment]
Updated description, moving to sdk core repo...
